HarvardChanSchool/canvas-theme
# Fix issue with Canvas transcripts not loading - CAN-29

Description
----------------

Adds a timeout function to Canvas pages if the div needed to place the transcript hasn't loaded ahead of the script.

Requirements
------------

Canvas access

**Branches:**


1. This branch

Test Steps
----------


1. Open up a Canvas course page in production, such as this one: [https://canvas.harvard.edu/courses/88461/pages/1-dot-1-motivation-why-regression-modeling](https://canvas.harvard.edu/courses/88461/pages/1-dot-1-motivation-why-regression-modeling). The transcript may load, or it may not. Refresh to view an instance where it will not load (try a private browsing window if you're continually seeing it load):
<img width="950" alt="Screen Shot 2022-11-15 at 11 40 50 AM" src="https://user-images.githubusercontent.com/10869319/201976364-5a8f84eb-d686-4ce7-a155-f74e874c44f0.png">

2. Open that same Canvas course in dev, by replacing the first part of the URL w/ https://harvard.test.instructure.com/, like so: [https://harvard.test.instructure.com/courses/88461/pages/1-dot-1-motivation-why-regression-modeling](https://harvard.test.instructure.com/courses/88461/pages/1-dot-1-motivation-why-regression-modeling). The page should load with the transcript. 
<img width="924" alt="Screen Shot 2022-11-15 at 11 42 21 AM" src="https://user-images.githubusercontent.com/10869319/201976751-ca944def-e02e-4037-b998-c0c07479d799.png">

3. Refresh the dev page multiple times to make sure the transcript always loads. 



Web Accessibility 

----------------


- [x] All changes in this PR conform with the current version of our [Web Accessibility PR Checklist](https://wiki.harvard.edu/confluence/pages/viewpage.action?spaceKey=HSPHWeb&title=Web+Accessibility+PR+Checklist).


Related
-----------------
**Jira Issues**



1. CAN-29


**ServiceNow tickets**

  
1. INC04716840
